### DISCUSSION TITLE
🙏 Help: Thymeleaf with KoliBri + Spring Boot - Submit Forms

### DISCUSSION THREAD

**Discussion by deleonio** [public-ui/kolibri#6700] [Q&A]
### Summary

I'm working in a project with Thymeleaf with KoliBri + Spring Boot, and I follow the steps to submit form and try to get the value of submit, but don't have, when you can, give me information how I can handle with value during submit please.

I'm using `kol-form`, `kol-inputs` and `kol-button`, get the form and do assignment in `kol-form._on` property to execute the callback. Thanks in advance for your help.

**Comment by deleonio** [comment]
Thank you for your interesting question.

Unfortunately, we have not yet had a Thymeleaf scenario with KoliBri. But I believe that here, as with PHP or JSF, the HTML is rendered from the server and sent to the browser. Is that correct?

If so, then it is probably a static scenario. Here we recommend activating the [experimental mode](https://public-ui.github.io/en/docs/get-started/first-steps#experimental-mode--lab) first. This manages the value handling in forms.

If possible, could you provide a reproduction example?

The form structure should look schematically as follows:

```
<html>
  <head>
    <meta name="kolibri" content="experimental-mode=true" />
  </head>
  <body>
    <form method="post" action="...">
      <kol-input-text _label="firstname"></kol-input-text>
      <kol-button _label="Submit" _type="submit"></kol-button>
    </form>
  </body>
</html>
```

FYI - @sdvg: Do you have any additions here?

**Comment by sdvg** [comment]
To me it's unclear whether this is about static form submission (i.e. handling the form data on the server) or using a client side callback function (as mentioned with the "`kol-form._on` property"). Could we clarify the scenario and maybe see some sample code?

**Comment by michaelodeus** [comment]
Hi, thanks for the help.
This is my first contact with KoliBri, so I did some tests with both scenarios `kol-form._on` and form but didn't work after I added  ` <meta name="kolibri" content="experimental-mode=true" />`  tag, it's worked.

```
<form id="loginForm" th:action="@{/login}" method="post">
      <kol-input-text _id="username" _type="text" _label="Username" _name="username"></kol-input-text>

      <kol-input-password _id="password" _label="Passwort" _name="password" ></kol-input-password>

      <kol-button  class="w-100" _type="submit" _label="Login" _variant="primary"></kol-button>
</form>

<script>
        const loginForm = document.querySelector("#loginForm");
        loginForm.addEventListener("submit", (event) => {
            console.log(event, "submit");
        });
</script>
```

**Reply by michaelodeus** [reply]
After adding this tag worked, but I received this message in the browser console.

What are the risks of using experimental mode in the real application?

![image](https://github.com/user-attachments/assets/dd54ee7a-8cd8-4ff6-af2a-bb75d697b7c7)

**Comment by deleonio** [comment]
Hi, please be patient. We had a sprint change today and haven't had time to respond yet.

Please check if you have duplicate fields in the DOM. Could you provide us with an excerpt from the browser's HTML?

We will then check if the duplicate rendering of IDs is caused by KoliBri.

**Reply by deleonio** [reply]
We analyze the redundant ID and will optimize that: https://github.com/public-ui/kolibri/issues/6706

**Reply by michaelodeus** [reply]
Hi, thanks for the answer and help.
In the DOM I can see two inputs with the same IDs, I attached an excerpt of `kol-input-text` generation for username input.

![image](https://github.com/user-attachments/assets/c3ba04b5-76d4-4720-8e15-0f16d0252bb8)

**Comment by deleonio** [comment]
@michaelodeus: I think the feature associated form fields is ready and we will discuss in the team to make this feature productive.

@sdvg: Let's complete the feature for the next sprint.

**Reply by michaelodeus** [reply]
Okay, thanks for all your hard work.
